syntax = "proto3";

service NotificationService {
  rpc SendEmail (EmailSendRequest) returns (EmailSendResponse);
  rpc SendEmailSimple (EmailSendRequest) returns (SimpleMessageResponse);

  rpc SendNotify (NotifyCreateRequest) returns (NotifyResponse);
  rpc SendNotifySimple (NotifyCreateRequest) returns (SimpleMessageResponse);
}

message EmailSendRequest {
  string sender = 1;
  string subject = 2;
  string message = 3;
  repeated string groups = 4;
  repeated string accounts = 5;
  repeated string emails = 6;
}

message EmailSendResponse {
  int64 id = 1;
  string sender = 2;
  string subject = 3;
  string message = 4;
  string status = 5;
  repeated Recipient recipients = 6;
  float timestamp = 7;
}

message SimpleMessageResponse {
  string message = 1;
}

message Recipient {
  string group = 1;
  repeated Account accounts = 2;
}

message Account {
  string username = 1;
  string email = 2;
  string name = 3;
}

message NotifyCreateRequest {
  string sender = 1;
  string subject = 2;
  string message = 3;
  bool is_email = 4;
  bool is_message = 5;
  bool is_line = 6;
  repeated string groups = 7;
  repeated string accounts = 8;
  repeated string emails = 9;
}


message NotifyResponse {
  int64 id = 1;
  string sender = 2;
  string subject = 3;
  string message = 4;
  bool is_email = 5;
  bool is_message = 6;
  bool is_line = 7;
  bool email_result = 8;
  bool message_result = 9;
  bool line_result = 10;
  float timestamp = 11;
}